### PR TITLE
Add config parameter to Azure storage backend to allow specifying the ARM endpoint

### PR DESCRIPTION
**Pull request by jstoker**
This PR introduces a config parameter to allow specifying the ARM endpoint to use when using the Azure Storage backend to support Azure Stack/sovereign clouds.

### CODE REVIEW

File: physical/azure/azure_test.go [hunk 1]
```diff
@@ -16,11 +16,14 @@ import (
...
 
-func environmentForCleanupClient(name string) (azure.Environment, error) {
-	if name == "" {
-		return azure.EnvironmentFromName("AzurePublicCloud")
+func environmentForCleanupClient(name string, armUrl string) (azure.Environment, error) {
+	if name != "" {
+		return azure.EnvironmentFromName(name)
 	}
-	return azure.EnvironmentFromName(name)
+	if armUrl != "" {
```
Comment (kalafut) [inline physical/azure/azure_test.go]: Should this check come first (for consistency with `azure.go`)?

**Comment by kalafut** [comment]
@jstoker Thanks for the PR! I think it looks pretty good. Just one question on the test.

File: physical/azure/azure_test.go (hunk 1 above)
Comment (jstoker) [reply physical/azure/azure_test.go]: Good point, I've adjusted the check to be similar to the one in `azure.go`.

**Comment by kalafut** [comment]
Thanks!

**Pull request merged**